lib/state.go

```diff
@@ -67,8 +67,22 @@ type State struct {
 	// TODO: maybe use https://golang.org/pkg/sync/#Pool ?
 	BPool *bpool.BufferPool
 
-	Vu, Iteration int64
-	Tags          map[string]string
+	Vu                   uint64
+	iteration            int64
+	Tags                 map[string]string
+	ScenarioName         string
+	scenarioVUID         map[string]uint64
+	IncrScIter           func() int64
+	IncrScIterGlobal     func() int64
+	scenarioVUIter       map[string]int64
+	scIter, scIterGlobal int64
+}
+
+// Init initializes some private state fields.
+func (s *State) Init() {
+	s.scenarioVUID = make(map[string]uint64)
+	s.scenarioVUIter = make(map[string]int64)
+	s.iteration, s.scIter, s.scIterGlobal = -1, -1, -1
```

Comment:
This is... nasty. I'd much prefer it if we could keep these as `uint64`, but since the increment needs to happen before `runFn()`, this seemed like the path of least resistance that doesn't involve a lot of additional checks.

Any better ideas?

Reply:
I guess they can be `^0` and adding 1 will overflow them ? does that have some problems 
https://play.golang.org/p/71mrOn9ATxg
you need the cast as otherwise the magical constant magic in go doesn't translate it to uint64 before `^` and instead it things it is the int `-1` :facepalm:

Reply:
Hhmm could be, I didn't think of that :smile: :+1:

Is relying on int overflow a valid pattern in Go?

Reply:
Resolved by 6fc76f85.

BTW @MStoykov is there a reason you used `^(uint64)(0)` instead of `^uint64(0)`?